assets/js/theme/custom/mega-menu.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Open mega menu on search button click
+ */
+
+import { constants } from './constants'
+
+export default function megaMenu() {
+    const { MOBILE_WIDTH } = constants
+    const dropdown = document.querySelector('.js-dropdown')
+    const searchButton = document.querySelector('.js-mega-menu-search')
+    const searchInput = document.querySelector('.js-search-quick')
+    const showMoreTriggers = document.querySelectorAll('.js-show-more-item')
+    let hasAttachedEvent = false
+
+    const openDesktopMenu = () => {
+        dropdown.classList.toggle('is-open')
+        searchInput.focus()
+    }
+
+    const updateEventListeners = () => {
+        const windowWidth = window.innerWidth
+        if (searchButton) {
+            if (windowWidth >= MOBILE_WIDTH && !hasAttachedEvent) {
+                searchButton.addEventListener('click', openDesktopMenu)
+                hasAttachedEvent = true
+            }
+        }
+    }
+
+    updateEventListeners()
+
+    window.addEventListener('resize', updateEventListeners)
+
+    showMoreTriggers.forEach((item) => {
+        const button = item.querySelector('.js-show-more')
+        if (button && !button.hasAttribute('data-event-added')) {
+            button.addEventListener('click', () => {
+                const categoryId = button.getAttribute('data-show-more')
+                const listItems = document.querySelectorAll(`.js-navigation-list-item[data-more="${categoryId}"]`)
+                button.classList.toggle('is-open')
+
+                const buttonLabels = button.querySelectorAll('.js-show-label')
+                if (listItems.length > 0) {
+                    listItems.forEach((listItem) => {
+                        listItem.classList.toggle('is-hidden')
+                    })
+
+                    buttonLabels.forEach((label) => {
+                        label.classList.toggle('u-hidden')
+                    })
+                }
+            })
+
+            button.setAttribute('data-event-added', 'true')
```

Comment:
What's the purpose of `data-event-added` ?

Reply:
So that the event listener is not added twice to the button

Reply:
But is it even possible that the event will be added twice? You call `megaMenu()` once, so you don't really need to worry about it in this case?

Also, consider wrapping `showMoreTriggers.forEach ....` code with `function`, it will look more clear :)

Reply:
Well not sure why when I hard refresh or when I resize it seems like the event listener is added twice and it doesn't let me "hide" the menu items, because it is toggling 'is-hidden' twice. Meaning it is removing and then adding the classname again. It means the on click event is twice in the button element. You can give it a try yourself without that 'date-event-added' code

Reply:
That's something we may need to investigate, as it could cause troubles in the future. But, even tho, why can't you just create a variable outside `megaMenu` called `isShowMoreEventAttached = false` and then assign it to true after finish iterating through forEach loop? Seems like much cleaner approach, than setting `data-event-added` attribute